lib/beaker/host.rb

```diff
@@ -370,7 +370,7 @@ def mkdir_p dir
         cmd = "if not exist #{windows_dirstring} (md #{windows_dirstring})"
       end
 
-      result = exec(Beaker::Command.new(cmd), :acceptable_exit_codes => [0, 1])
+      result = exec(Beaker::Command.new(cmd), :acceptable_exit_codes => [0])
```

Comment:
Before we were wrapping a failure so that we could return the correct exit code for the caller to do with as they please - I'm not sure about changing the logic here so that failures raise exceptions.  Maybe we should add an opt for 'report_failure' or 'raise_failure' ?